common/lib/dependabot/file_fetchers/base.rb

```diff
@@ -861,18 +858,40 @@ def decode_binary_string(str)
 
       sig { params(path: String).returns(T::Array[String]) }
       def find_submodules(path)
-        SharedHelpers.run_shell_command(
-          <<~CMD
-            git -C #{path} ls-files --stage
-          CMD
-        ).split("\n").filter_map do |line|
+        lfs_enabled = is_lfs_enabled(path) if lfs_enabled.nil?
+        SharedHelpers.run_shell_command("git-lfs-checkout") if lfs_enabled
+        command_string = get_command_string(path, lfs_enabled)
+        #  eep command_string
+        SharedHelpers.run_shell_command(command_string).split("\n").filter_map do |line|
           info = line.split
 
           type = info.first
           path = T.must(info.last)
-
           next path if type == DependencyFile::Mode::SUBMODULE
         end
+      rescue SharedHelpers::HelperSubprocessFailed => e
+        Dependabot.logger.warn("LFS is enabled in this repo.  Please use an LFS enabled client") if lfs_enabled
+        Dependabot.logger.error(e.message)
+        raise e.exception("Message: #{e.message}")
+      end
+
+      sig { params(path: String).returns(T.nilable(T::Boolean)) }
+      def is_lfs_enabled(path)
+        filepath = File.join(path, ".gitattributes")
+        lfs_enabled = T.let(true, T::Boolean) if File.exist?(filepath) && File.readable?(filepath) &&
+          SharedHelpers.run_shell_command("cat #{filepath} | grep \"filter=lfs\"").include?("filter=lfs")
+      rescue Exception => e
+        # this should not be needed, but I don't trust 'should'
+        lfs_enabled = T.let(false, T::Boolean)
+        raise e.exception("Message: #{e.message}")
+      end
```

Comment:
I think all of this can be simplified to this:

```suggestion
sig { params(path: String).returns(T::Boolean) }
  def lfs_enabled?(path)
    gitattributes_path = File.join(path, '.gitattributes')

    return false unless File.exist?(gitattributes_path)

    File.readlines(gitattributes_path).each do |line|
      return true if line =~ /filter=lfs/
    end

    false
  end```